assets/src/hooks/useRoutePatternsById.ts

```diff
@@ -9,10 +9,18 @@ const useRoutePatternsById = (
     null
   )
   useEffect(() => {
+    let canceled = false
     if (routeId === null) {
       setRoutePatterns([])
     } else {
-      fetchRoutePatterns(routeId).then(setRoutePatterns)
+      fetchRoutePatterns(routeId).then((routePatterns) => {
+        if (!canceled) {
+          setRoutePatterns(routePatterns)
+        }
+      })
+    }
+    return () => {
+      canceled = true
```

Comment:
**suggestion:** might be worth adding a test for the race condition as well if it isn't too difficult.

Reply:
Test added in [192c2ad](https://github.com/mbta/skate/pull/1904/commits/192c2ad935593841c3e2a862f9bd487f16342383) - I don't love the way it relies on the `patternsFor66DidResolve` variable though.